src/codegen/baseline_jit.h

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) 2014-2015 Dropbox, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PYSTON_CODEGEN_BASELINEJIT_H
+#define PYSTON_CODEGEN_BASELINEJIT_H
+
+#include <llvm/ADT/ArrayRef.h>
+
+#include "asm_writing/rewriter.h"
+#include "codegen/ast_interpreter.h"
+#include "gc/heap.h"
+
+namespace pyston {
+
+#define ENABLE_BASELINEJIT_ICS 1
+
+class AST_stmt;
+class Box;
+class BoxedDict;
+class BoxedList;
+class BoxedTuple;
+
+class SetGlobalIC;
+class GetGlobalIC;
+class SetAttrIC;
+class GetAttrIC;
+class SetItemIC;
+class GetItemIC;
+class CompareIC;
+class AugBinopIC;
+class UnaryopIC;
+class RuntimeCallIC;
+
+class JitFragment;
+
+// Manages a fixed size memory block which stores JITed code.
```

Comment:
Minor request -- I think it'd be nice to start with a little bit more detail here about the overall baseline-JIT strategy and how it relates to the interpreter.  Just a quick high-level overview of what the classes in this file are for, how they get used, what our strategy is for going back and forth between the interpreter/jitted code.